api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java

```diff
@@ -870,12 +869,95 @@ public void getDataDictionaryEntry() {
     verify(mockDSDataDictionaryDao, times(1)).findFirstByFieldNameAndDomain("MockValue", "PERSON");
   }
 
+  @Test
+  public void testWgsCohortExtraction_permissions() {
+    cdrVersion.setWgsBigqueryDataset("wgs");
+    cdrVersionDao.save(cdrVersion);
+
+    DataSet dataSet =
+        dataSetController
+            .createDataSet(
+                workspace.getNamespace(), workspace.getName(), buildValidDataSetRequest())
+            .getBody();
+
+    when(fireCloudService.getWorkspace(workspace.getNamespace(), workspace.getName()))
+        .thenReturn(new FirecloudWorkspaceResponse().accessLevel("NO ACCESS"));
+    assertThrows(
+        ForbiddenException.class,
+        () -> {
+          dataSetController.extractCohortGenomes(
+              workspace.getNamespace(), workspace.getName(), dataSet.getId());
+        });
+
+    when(fireCloudService.getWorkspace(workspace.getNamespace(), workspace.getName()))
+        .thenReturn(new FirecloudWorkspaceResponse().accessLevel("READER"));
+    assertThrows(
+        ForbiddenException.class,
+        () -> {
+          dataSetController.extractCohortGenomes(
+              workspace.getNamespace(), workspace.getName(), dataSet.getId());
+        });
+
+    when(fireCloudService.getWorkspace(workspace.getNamespace(), workspace.getName()))
+        .thenReturn(new FirecloudWorkspaceResponse().accessLevel("WRITER"));
+    dataSetController.extractCohortGenomes(
+        workspace.getNamespace(), workspace.getName(), dataSet.getId());
+
+    when(fireCloudService.getWorkspace(workspace.getNamespace(), workspace.getName()))
+        .thenReturn(new FirecloudWorkspaceResponse().accessLevel("OWNER"));
+    dataSetController.extractCohortGenomes(
+        workspace.getNamespace(), workspace.getName(), dataSet.getId());
+
+    when(fireCloudService.getWorkspace(workspace.getNamespace(), workspace.getName()))
+        .thenReturn(new FirecloudWorkspaceResponse().accessLevel("PROJECT_OWNER"));
+    dataSetController.extractCohortGenomes(
+        workspace.getNamespace(), workspace.getName(), dataSet.getId());
+  }
+
+  @Test
+  public void testWgsCohortExtraction_notFound() {
+    assertThrows(
+        BadRequestException.class,
+        () -> {
+          dataSetController.extractCohortGenomes(
+              workspace.getNamespace(), workspace.getName(), 404L);
+        });
+  }
+
+  @Test
+  public void testWgsCohortExtraction_validCdr() throws Exception {
+    DataSet dataSet =
+        dataSetController
+            .createDataSet(
+                workspace.getNamespace(), workspace.getName(), buildValidDataSetRequest())
+            .getBody();
+
```

Comment:
nit: can you add a line here to unset the wgs field? makes it clearer what "validCdr" is testing against and also makes the test resilient to a change in the fixture data.

Reply:
Done